utils/id.go

```diff
@@ -41,3 +44,61 @@ func LocalNodeID() (string, error) {
 	}
 	return fmt.Sprintf("%s%s", NodePrefix, HashedID(hostname)[:8]), nil
 }
+
+var ErrInvalidGuid = errors.New("invalid guid")
+
+var b62Index = newB62Index()
+var b62Chars = []byte(shortuuid.DefaultAlphabet)
+
+func newB62Index() [256]byte {
+	var index [256]byte
+	for i := 0; i < len(shortuuid.DefaultAlphabet); i++ {
+		index[shortuuid.DefaultAlphabet[i]] = byte(i)
+	}
+	return index
+}
+
+func guidPrefix[T livekit.Guid]() string {
+	var id T
+	switch any(id).(type) {
+	case livekit.TrackID:
+		return TrackPrefix
+	case livekit.ParticipantID:
+		return ParticipantPrefix
+	case livekit.RoomID:
+		return RoomPrefix
+	case livekit.NodeID:
+		return NodePrefix
+	default:
+		panic("unreachable")
+	}
+}
+
+func MarshalGuid[T livekit.Guid](id T) livekit.GuidBlock {
+	var b livekit.GuidBlock
+	prefix := guidPrefix[T]()
+	idb := []byte(id)[len(prefix):]
+	for i := 0; i < 12; i += 4 {
+		j := (i * 6) >> 3
+		b[j] |= b62Index[idb[i]]<<2 | b62Index[idb[i+1]]>>4
```

Comment:
I am probably not reading this properly. Is there a need to do `|=` here? Seems like each byte of `b` is just set once.

Reply:
oh, good catch. it looks like i forgot to change this when i unrolled these